inc/fields/class-field-post-select.php

```diff
@@ -0,0 +1,167 @@
+	}
+
+	private function setup_actions() {
+
+		add_filter( 'shortcode_ui_fields', array( $this, 'filter_shortcake_fields' ) );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'action_admin_enqueue_scripts' ), 100 );
+		add_action( 'wp_ajax_shortcake_post_field', array( $this, 'action_wp_ajax_shortcake_post_field' ) );
+		add_action( 'shortcode_ui_loaded_editor', array( $this, 'action_shortcode_ui_loaded_editor' ) );
+
+	}
+
+	public function filter_shortcake_fields( $fields ) {
+		return array_merge( $fields, $this->fields );
+	}
+
+	public function action_admin_enqueue_scripts() {
+
+		$plugin_dir =  dirname( dirname( __FILE__ ) );
+
+		wp_enqueue_script(
+			'shortcake-field-post-select',
+			plugins_url( '/js/field-post-select.js', $plugin_dir ),
+			array( 'shortcode-ui', 'select2' )
+		);
+
+		wp_enqueue_script( 'select2', plugins_url( 'lib/select2/select2.min.js', $plugin_dir ) , array( 'jquery', 'jquery-ui-sortable' ), '3.5.2' );
+		wp_enqueue_style( 'select2', plugins_url( 'lib/select2/select2.css', $plugin_dir ), null, '3.5.2' );
+
+		wp_localize_script( 'shortcake-field-post-select', 'shortcakePostFieldData', array(
+			'nonce' => wp_create_nonce( 'shortcake_field_post_select' ),
+		) );
+
+	}
+
+	/**
+	 * Output styles and templates used by post select field.
+	 */
+	// public function action_print_media_templates() {
+	public function action_shortcode_ui_loaded_editor() {
+
+		?>
+
+		<style>
+
+			.edit-shortcode-form .select2-container {
+				min-width: 300px;
+			}
+
+			.edit-shortcode-form .select2-container a {
+				transition: none;
+				-webkit-transition: none;
+			}
+
+			.wp-admin .select2-drop {
+				z-index: 160001;
+			}
+
+		</style>
+
+		<script type="text/html" id="tmpl-shortcake-field-post-select">
+			<div class="field-block">
+				<label for="{{ data.attr }}">{{ data.label }}</label>
+				<input type="text" name="{{ data.attr }}" id="{{ data.attr }}" value="{{ data.value }}" class="shortcake-post-select" />
+			</div>
+		</script>
+
+		<?php
+	}
+
+	/**
+	 * Ajax handler for select2 post field queries.
+	 * Output JSON containing post data.
+	 * Requires that shortcode, attr and nonce are passed.
+	 * Requires that the field has been correctly registred and can be found in $this->post_fields
+	 * Supports passing page number and search query string.
+	 *
+	 * @return null
+	 */
+	public function action_wp_ajax_shortcake_post_field() {
+
+		$nonce               = isset( $_GET['nonce'] ) ? sanitize_text_field( $_GET['nonce'] ) : null;
+		$requested_shortcode = isset( $_GET['shortcode'] ) ? sanitize_text_field( $_GET['shortcode'] ) : null;
+		$requested_attr      = isset( $_GET['attr'] ) ? sanitize_text_field( $_GET['attr'] ) : null;
+		$response            = array( 'posts' => array(), 'found_posts'=> 0, 'posts_per_page' => 0 );
+
+		$shortcodes = Shortcode_UI::get_instance()->get_shortcodes();
+
+		if ( ! wp_verify_nonce( $nonce, 'shortcake_field_post_select' ) ) {
+			wp_send_json_error( $response );
+		}
```

Comment:
It looks like `wp_ajax_find_posts()` doesn't have a capability check, so we're good.